File: контест 1/A.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;

int
main(void){
    int n;
    cin >> n;
    double left_1 = numeric_limits<double>::infinity();
    double right_1 = -numeric_limits<double>::infinity();
    double left_2 = numeric_limits<double>::infinity();
    double right_2 = -numeric_limits<double>::infinity();
    double highest_right = -numeric_limits<double>::infinity();
    double highest_left = -numeric_limits<double>::infinity();
    for (int i = 0; i < n; i++){
        string myString;
        cin.ignore();
        getline(cin, myString);
        stringstream iss(myString);
        double tmp;
        vector<double> arg;
        while ( iss >> tmp ){
            arg.push_back(tmp);
        }
        if (abs(arg[1]) > 0 && (arg[0] > 0)){
            highest_right = max(highest_right, abs(arg[1]));
            cout << arg[1] << arg[0] << endl;
        }
        else if (abs(arg[1]) > 0 && (arg[0] < 0)){
            highest_left = max(highest_left, abs(arg[1]));
            cout << 1;
        }
        if ((arg[1] == 0) && (arg[0] > 0)){
            if (arg[0] < left_1){
                if (left_1 != numeric_limits<double>::infinity()){
                    right_1 = max(left_1, right_1);
                }
                left_1 = arg[0];
            }
            else if (arg[0] > right_1){
                if (right_1 != -numeric_limits<double>::infinity()){
                    left_1 = min(left_1, right_1);
                }
                right_1 = arg[0];
            }
        }
        if ((arg[1] == 0) && (arg[0] < 0)){
            if (arg[0] > left_2){
                if (left_2 != -numeric_limits<double>::infinity()){
                    right_2 = max(left_2, right_2);
                }
                left_2 = arg[0];
            }
            else if (arg[0] < right_2){
                if (right_2 != numeric_limits<double>::infinity()){
                    left_2 = min(left_2, right_2);
                }
                right_2 = arg[0];
            }
        }
    }
    double s_1 = abs((right_1 - left_1) * highest_right / 2);
    double s_2 =  abs((right_2 - left_2) * highest_left / 2);
    if (max(s_1, s_2) != numeric_limits<double>::infinity()){
        cout << max(s_1, s_2) << endl;;
    }
    else if (min(s_1, s_2) != numeric_limits<double>::infinity()){
        cout << min(s_1, s_2) << endl;;
    }
    else{
        cout << 0 << endl;;
    }
    cout << right_2 << left_2 << highest_left << endl;
}

File: контест 1/A.py
import sys
input = sys.stdin.buffer.readline

def read_int():
    return int(input())


left_1, left_2, right_1, right_2 = float('inf'), float('inf'), -float('inf'), -float('inf')
high_right, high_left = -float('inf'), -float('inf')
n = read_int()
for _ in range(n):
    x, y = map(float, input().split())


    if (abs(y) > 0 and x > 0):
        high_right = max(high_right, abs(y))
    elif (abs(y) > 0 and x < 0):
        high_left = max(high_left, abs(y))
    if (y == 0) and (x > 0):
        if (x < left_1):
            if (abs(left_1) != float('inf')):
                right_1 = max(right_1, left_1)
            left_1 = x
        elif (x > right_1):
            if (abs(right_1) != float('inf')):
                left_1 = min(right_1, left_1)
            right_1 = x
    elif (y == 0) and (x < 0):
        if (x > right_2):
            if (abs(right_2) != float('inf')):
                left_2 = min(right_2, left_2)
            right_2 = x
        elif (x < left_2):
            if (abs(left_2) != float('inf')):
                right_2 = max(right_2, left_2)
            left_2 = x
s_1 = abs((right_1 - left_1) * high_right / 2)
s_2 = abs((left_2 - right_2) * high_left / 2)
if min(s_1, s_2) == float('inf'):
    print(0)
else:
    if max(s_1, s_2) == float('inf'):
        print(min(s_1, s_2))
    else:
        print(max(s_1, s_2))

sys.stdout.flush()
